js/content/store.js

```diff
@@ -3119,9 +3174,138 @@ let WishlistPageClass = (function(){
                 DynamicStore.clear();
                 location.reload();
             });
-        }, true)
+        }, true);
     }
 
+    WishlistPageClass.prototype.showModalDialog = function() {
+        // Partly copied from shared_global.js
+        ExtensionLayer.runInPageContext(`function() {
+            let deferred = new jQuery.Deferred();
+            let fnOK = () => deferred.resolve();
+
+            let Modal = _BuildDialog(
+                "${Localization.str.export_wishlist}",
+                \`${this.exportModalTemplate}\`,
+                [], fnOK);
+            deferred.always(() => Modal.Dismiss());
+
+            Modal.m_fnBackgroundClick = () => Modal.Dismiss();
+            Modal.Show();
+
+            $J("#es_export_modal input[placeholder]").each(function() {
+                $J(this).attr('size', $J(this).attr('placeholder').length);
+            });
+
+            // attach the deferred's events to the modal
+            deferred.promise(Modal);
+        }`);
+
+        document.addEventListener("click", clickListener);
+
+        function clickListener(e) {
+            if (e.target.closest(".es_export_modal_submit")) {
+                e.preventDefault();
+                exportWishlist();
+                ExtensionLayer.runInPageContext(() => CModal.DismissActiveModal());
+            } else if (e.target.closest(".es_export_modal_close")) {
+                ExtensionLayer.runInPageContext(() => CModal.DismissActiveModal());
+            } else {
+                return;
+            }
+            document.removeEventListener("click", clickListener);
+        }
+
+        function exportWishlist() {
+            ExtensionLayer.runInPageContext(`function(){
+                let data = { rgAllApps: g_Wishlist.rgAllApps, g_rgAppInfo, options: {} };
+                $J("#es_export_form").serializeArray().forEach((item) => data.options[item.name] = item.value);
+                Messenger.postMessage("exportWishlist", JSON.stringify(data));
+            }`);
+            
+            Messenger.addMessageListener("exportWishlist", (data) => {
+                if (!data) { return; }
+                let { rgAllApps, g_rgAppInfo, options } = JSON.parse(data);
+                if (!rgAllApps || !g_rgAppInfo || !options) { return; }
+
+                let toexport;
+                let notes = SyncedStorage.get("user_notes") || {};
+                let json = rgAllApps.map(appid => {
+                    let appinfo = g_rgAppInfo[appid];
+                    appinfo.appid = appid;
+                    if (notes[appid]) {
+                        appinfo.note = notes[appid];
+                    }
+                    return appinfo;
+                });
+
+                if (options.export_type === "JSON") {
+                    toexport = JSON.stringify(json, null, 4);
```

Comment:
This will export all available information from `g_rgAppInfo[appid]`, which, if you take a closer look, varies in usefulness (e.g. it will export HTML elements).
I think some options just like the plain text export could prove useful here

Reply:
Since it's JSON anyway, I feel like users (devs) can shape it how they want it themselves.

Reply:
Ok, yeah, you're right